backend/drm/drm.c

```diff
@@ -649,6 +654,8 @@ static bool drm_connector_set_cursor(struct wlr_output *output,
 		enum wl_output_transform transform,
 		int32_t hotspot_x, int32_t hotspot_y, bool update_texture) {
 	struct wlr_drm_connector *conn = get_drm_connector_from_output(output);
+	assert((conn->state != WLR_DRM_CONN_LEASED || texture == NULL)
+			&& "Use of leased connector is a programming error");
```

Comment:
This `texture == NULL` exception is necessary because when the output is destroyed, its cursors are destroyed as well.